resources/ext/config/conf.d/puppetserver.conf

```diff
@@ -2,7 +2,7 @@
 jruby-puppet: {
     # Where the puppet-agent dependency places puppet, facter, etc...
     # Puppet server expects to load Puppet from this location
-    ruby-load-path: [/opt/puppetlabs/puppet/lib/ruby/vendor_ruby]
+    ruby-load-path: [/opt/puppetlabs/puppet/lib/ruby/vendor_ruby, /opt/puppetlabs/puppet/cache/lib]
```

Comment:
@joshcooper I understand that this is distasteful.  Is it a new issue with Puppet 4, though, or did we have this problem in the past as well?  Do you have an alternative suggestion for how hiera backends in modules could be loaded by the master if we don't use the pluginsync / agent path as has apparently been done in the past?